pkg/machine/qemu/machine.go

```diff
@@ -762,15 +803,14 @@ func (v *MachineVM) ProxyPID() (int, error) {
 	return proxyPid, nil
 }
 
-// cleanupVMProxyProcess kills the proxy process and removes the VM's pidfile
-func (v *MachineVM) cleanupVMProxyProcess(proxyProc *os.Process) error {
+// cleanupVMProxyProcess kills the proxy process
+func (v *MachineVM) cleanupVMProxyProcess(proxyPid int) error {
```

Comment:
out of curiosity, why did you choose to change the type here from something reasonably strong to something more generic like an int?  im not sure i have an objection with it, i would however like to know.

Reply:
Because internally this `os.Process` pointer was used only to call `Kill` method https://pkg.go.dev/os#Process.Kill, which really doesn't work as we need to send SIGTERM for correct cleanup and it also doesn't work on Windows. There is no point extracting the pid here from `os.Proccess` as previously we acquired that looking up pid value we already stored as part of the state. `os.Process` really brings no benefits there if we can't use its API.